litmus/AArch64Compile_litmus.ml

```diff
@@ -431,7 +441,32 @@ module Make(V:Constant.S)(C:Config) =
           memo=sprintf "%s %s,%s" memo f1 f2;
           inputs = r2; outputs=r1; reg_env=add_q (r1@r2);}
 
+    let do_mov memo v rd k os = match v, k, os with
+    | V32, K k, Some(LSL(s)) ->
+        let r1,f1 = arg1 "wzr" (fun s -> "^wo"^s) rd in
+        { empty_ins with
+          memo=sprintf "%s %s, #%d, LSL #%d" memo f1 k s;
+          outputs=r1; reg_env=add_w r1;}
+    | V32, K k, None ->
+        let r1,f1 = arg1 "wzr" (fun s -> "^wo"^s) rd in
+        { empty_ins with
+          memo=sprintf "%s %s, #%d" memo f1 k;
+          outputs=r1; reg_env=add_w r1;}
+    | V64, K k, Some(LSL(s)) ->
+        let r1,f1 = arg1 "xzr" (fun s -> "^o"^s) rd in
+        { empty_ins with
+          memo=sprintf "%s %s, #%d, LSL #%d" memo f1 k s;
+          outputs=r1; reg_env=add_q r1;}
+    | V64, K k, None ->
+        let r1,f1 = arg1 "xzr" (fun s -> "^o"^s) rd in
+        { empty_ins with
+          memo=sprintf "%s %s, #%d" memo f1 k;
+          outputs=r1; reg_env=add_q r1;}
+    | _ -> Warn.fatal "Illegal form of %s instruction" memo
+
```

Comment:
Barrel shifter argument handling could probably be factored out. This will prove convenient if one add the other variants (LSR etc)  later.

Reply:
good idea, will fix